flowtorch/bijectors/compose.py

```diff
@@ -31,38 +32,42 @@ def __init__(
         self.domain = self.bijectors[0].domain  # type: ignore
         self.codomain = self.bijectors[-1].codomain  # type: ignore
 
-        # Make parameters accessible to dist.Flow
-        self._params = torch.nn.ModuleList(
-            [
-                b._params  # type: ignore
-                for b in self.bijectors
-                if isinstance(b._params, torch.nn.Module)  # type: ignore
-            ]
-        )
-
         self._context_shape = context_shape
 
-    # NOTE: We overwrite forward rather than _forward so that the composed
-    # bijectors can handle the caching separately!
-    def forward(self, x: torch.Tensor, context: torch.Tensor = None) -> torch.Tensor:
+    def parameters(self) -> Iterator[torch.Tensor]:
+        for b in self.bijectors:
+            for param in b.parameters():
+                yield param
+
+    def _forward(
```

Comment:
Should this not be `forward` rather than `_forward`?

Reply:
not sure
personally I think that we should interfere as rarely as possible with `forward` and `inverse` as we might change their behaviour in the future in the Bijector class. That means that we'll need to go through every module and check if they re-implement `forward` or not. The current solution I propose deals well with `_forward`. 
What do you think?

Reply:
Ah yes I agree with you! Let's keep the current solution that overrides `_forward` and `_inverse`. That way we won't have to reimplement the `BijectiveTensor` code again here